### ISSUE TITLE
No output

### ISSUE THREAD

**Issue by rctneil** [brestolho/npm-bowerFixCssPath#1]
Hey, This module is exactly what I need but I'm not seeing any results.  Any ideas? My code is below:

```
var config = {
  "debug": false,
  "absolutePath": "./public/assets/",
  "types":{
    "fonts":{
      extensions: [".eot", ".woff", ".ttf", ".woff2"],
      prefixPath: "fonts2/"
    },
    "imgs":{
      extensions: [".png", ".jpg", ".gif", ".jpeg"],
      prefixPath: "images2/"
    },
    "svgs":{
      extensions: [".svg"],
      prefixPath: "svgs2/"
    }
  }
};

gulp.task("bowerFixCssWithConfig", function() {
  return gulp.src('./public/assets/stylesheets/vendor.css')
    .pipe(bowerFixCss(config))
    .pipe(gulp.dest(function (file) {
      return file.base;
    }));
});

```
I've also just noticed that the paths in the .css file are being updated but to the following:
```
  .flag-icon-aq.flag-icon-squared {
    background-image: url('./public/assets/vendor.css/svgs2/aq.svg')
  }
```

What's going on here?

UPDATE:  I figured out that I need to use this alongside bower-installer. I tried it and bower-installer is moving the bower files (not to the right place yet but it's a start).

Is there a way of running bower-installer as part of the gulp task of bower-fix-css-paths rather than only from the command line?

**Comment by brestolho** [comment]
Hi Neil,

when i build this package, i was working with bower-installer and never
tried to test this package without it (shame on me).
Good news, it works without bower but i have small bug with "paths" in
source files.

You can fix this, just changing the line "return gulp.src('./public/assets/
stylesheets/vendor.css')" to "return gulp.src('./public/assets/**/*.css')".

Thank you to alert me with this one.

**Comment by rctneil** [comment]
Great! I'll give that a go a bit later. Did you see the updates I made to my OP?

Basically I would like this functionality overall:

Use Bower to download all my components into my bower-components directory.  Then a single Gulp task that will move the associated files into a new location in /public/assets (in images, svgs fonts directories etc)  then all the url paths in my CSS files to get updated to point to those new locations.

So basically need to know how to get the "move" features of bower-installer working as part of a gulp task?

Any suggestions?

**Comment by rctneil** [comment]
I'm a bit wary of changing the line you mentioned. Mainly as I only want to test things out using the vendor.css file and not all the .css files as your line would do.

**Comment by brestolho** [comment]
If is just for testing, you could put the vendor.css in a different folder.

I suggest you use "bower-installer" instead of "bower" to download your
components and organize them at the same time. In my case i use gulp to
execute the "bower-installer" by exec command but you can use it as you use
bower.

Don't know any package for gulp that does the same as bower-installer but
it's possible manually, maybe using  selectors like this
/**/*.[jpg|gif|woff|svg|...] and move all assets to your destiny folders.
To move files you need copy first and clean later. Like example here
<http://stackoverflow.com/questions/21546931/use-gulp-to-select-and-move-directories-and-their-files>,
to clean you can use "gulp-clean".

Don't recommend you to copy all your asset components to single folders eg:
[imgs, svgs, fonts, ...], because components can have assets with same
names and you will lost track of them.

Hope this help you out

**Comment by rctneil** [comment]
> 
> If is just for testing, you could put the vendor.css in a different folder.
- I'll try this.

> 
> I suggest you use "bower-installer" instead of "bower" to download your
> components and organize them at the same time. In my case i use gulp to
> execute the "bower-installer" by exec command but you can use it as you use
> bower.
- 1. How does bower-installer know what files to move? and 2. How do you run a command line script from gulp?

> 
> Don't know any package for gulp that does the same as bower-installer but
> it's possible manually, maybe using  selectors like this
> /**/*.[jpg|gif|woff|svg|...] and move all assets to your destiny folders.
> To move files you need copy first and clean later. Like example here
> <http://stackoverflow.com/questions/21546931/use-gulp-to-select-and-move-directories-and-their-files>,
> to clean you can use "gulp-clean".
> 
> Don't recommend you to copy all your asset components to single folders eg:
> [imgs, svgs, fonts, ...], because components can have assets with same
> names and you will lost track of them.
- This is very very true and I think I will need to have another look at my strategy.
> 
> Hope this help you out
- Thankyou!

**Comment by brestolho** [comment]
1. How does bower-installer know what files to move?

You need extra info in your bower.json, see [here](https://www.npmjs.com/package/bower-installer)
Here's an example:
```
{
    "name": "TEST",
    "version": "1.0",
    "dependencies": {
        "jquery": "2.2.*",
        "jqueryui": "1.11.*",
        "font-awesome": "4.*"
    },
    "install": {
        "base": "./vendor",
        "path": {
            "js": "{name}",
            "css": "{name}",
            "eot": "{name}/fonts",
            "svg": "{name}/fonts",
            "ttf": "{name}/fonts",
            "woff": "{name}/fonts",
            "woff2": "{name}/fonts",
            "otf": "{name}/fonts",
            "jpg": "{name}/img",
            "png": "{name}/img",
            "gif": "{name}/img",
            "jpeg": "{name}/img"
        },
        "sources": {
            "jquery": [
                "bower_components/jquery/dist/jquery.js",
                "bower_components/jquery/dist/jquery.min.js"
            ],
            "jqueryui": [
                "bower_components/jqueryui/jquery-ui.min.js",
                "bower_components/jqueryui/themes/smoothness/jquery-ui.min.css",
                "bower_components/jqueryui/themes/smoothness/images/*.*"
            ],
            "font-awesome": [
                "bower_components/font-awesome/css/font-awesome.min.css",
                "bower_components/font-awesome/fonts/*"
            ],
        }
    }
}
```
The "base" is your destiny folder, in the "path" you see eg: "{name}/img" (the name here is the name of the package, like "jqueryui") and the source are the rules of what to copy for each package.

2. How do you run a command line script from gulp?

```
gulp.task("bowerInstaller", function () {
    return run('bower-installer').exec();
});
```

**Comment by rctneil** [comment]
Fantastic. Will give this a go later!  I'll report back!
